### PR TITLE
Feature lwdev 4794 add models adapter

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Service.OperationsHistory/Services/HistoryFilterPredicates.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                    default: return false;
+                }
+            };
+        }
+
+        public static Func<HistoryOperation, bool> IfAssetEquals(string assetId)
+        {
+            return operation =>
+            {
+                if (string.IsNullOrWhiteSpace(assetId)) return true;
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Services/HistoryFilterPredicates.cs]: Is it correct that nothing==true?

File: src/Lykke.Service.OperationsHistory/Modules/ServiceModule.cs [hunk 2]
```diff
@@ -93,6 +105,28 @@ private void RegisterAzureRepositories(ContainerBuilder builder)
         private void RegisterServiceClients(ContainerBuilder builder)
         {
             builder.RegisterLykkeServiceClient(_settings.ClientAccountServiceClient.ServiceUrl);
+            builder.RegisterInstance<IAssetsService>(
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Modules/ServiceModule.cs]: Is it necessary to create an instance here? Why not just register like it done in the line 114

File: src/Lykke.Service.OperationsHistory/Models/HistoryOperationModel.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    public class CashOutHistoryOperationModel : BaseCashOperationModel
+    {
+        [JsonConverter(typeof(StringEnumConverter))]
+        public CashOutState CashOutState { get; set; }
+    }
+
+    public class TradeHistoryOperationModel
+    {
+        public string Id { get; set; }
+        public string DateTime { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Models/HistoryOperationModel.cs]: Why string type for the datetime field?

File: src/Lykke.Service.OperationsHistory/Models/HistoryOperationModel.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        public string Asset { get; set; }
+        public double Volume { get; set; }
+        public bool IsSettled { get; set; }
+        public string LimitOrderId { get; set; }
+        public string MarketOrderId { get; set; }
+        public string ContextOperationType { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Models/HistoryOperationModel.cs]: Can be here enum?

File: src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    public class CashOutHistoryOperation : BaseCashOperation
+    {
+        [JsonConverter(typeof(StringEnumConverter))]
+        public CashOutState CashOutState { get; set; }
+    }
+
+    public class TradeHistoryOperation
+    {
+        public string Id { get; set; }
+        public string DateTime { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs]: Why string here?

File: src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                DateTime = dateTime,
+                Id = id,
+                CashIn = cashIn,
+                CashOut = cashout,
+                Trade = trade
+            };
+        }
+    }
+
+    public class BaseCashOperation
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs]: Can be abstract

File: src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        public string Id { get; set; }
+        public double Amount { get; set; }
+        public string DateTime { get; set; }
+        public string Asset { get; set; }
+        public string BlockChainHash { get; set; }
+        public bool IsRefund { get; set; }
+        public string AddressFrom { get; set; }
+        public string AddressTo { get; set; }
+        public bool IsSettled { get; set; }
+        public string Type { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs]: Why not enum?

File: src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs (hunk 6 above)
Comment (tarurar) [reply src/Lykke.Service.OperationsHistory.Core/Domain/HistoryOperation.cs]: +

File: client/Lykke.Service.OperationsHistory.Client/Lykke.Service.OperationsHistory.Client.csproj [hunk 8]
```diff
@@ -16,16 +15,12 @@
...
-
   <ItemGroup>
     <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.2.0" />
-    <PackageReference Include="AutoMapper" Version="6.2.1" />
-    <PackageReference Include="Lykke.Common" Version="3.0.2" />
+    <PackageReference Include="Lykke.Common" Version="4.3.0" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.10" />
   </ItemGroup>
-
   <ItemGroup>
```
Comment (MikhailGerasimov) [inline client/Lykke.Service.OperationsHistory.Client/Lykke.Service.OperationsHistory.Client.csproj]: Enable XML documentations for the release configuration

**Pull request closed without merging**